src/main/java/org/mule/extension/email/internal/StoredEmailContentFactory.java

```diff
@@ -134,4 +162,27 @@ private static MediaType getMediaType(Message message) {
       return MediaType.TEXT;
     }
   }
+
+  private String getUniqueFileName(Set<String> keys, String fileName) {
+    if (keys.contains(fileName)) {
+      return resolveUniqueFileName(keys, fileName, 0);
+    }
+    return fileName;
+  }
+
+  private String resolveUniqueFileName(Set<String> keys, String fileName, Integer lastSuffixTried) {
+    String extension = "";
+    String baseName = fileName;
+    int lastIndexOf = fileName.lastIndexOf('.');
+    if (lastIndexOf != -1) {
+      extension = fileName.substring(fileName.lastIndexOf('.'));
+      baseName = fileName.substring(0, fileName.lastIndexOf('.'));
+    }
+
+    String candidateFileName = baseName + "_" + ++lastSuffixTried + extension;
+    if (keys.contains(candidateFileName)) {
+      return resolveUniqueFileName(keys, fileName, lastSuffixTried);
```

Comment:
Doing this will resolve the index of all over again, we should try only changing the suffix.